src/web/api/src/main/java/org/locationtech/geogig/rest/geopkg/GeoPkgExportWebOp.java

```diff
@@ -0,0 +1,215 @@
+import org.geotools.geopkg.GeoPkgDataStoreFactory;
+import org.geotools.jdbc.JDBCDataStore;
+import org.locationtech.geogig.api.Context;
+import org.locationtech.geogig.api.NodeRef;
+import org.locationtech.geogig.api.ObjectId;
+import org.locationtech.geogig.api.RevFeatureType;
+import org.locationtech.geogig.api.RevObject;
+import org.locationtech.geogig.api.RevTree;
+import org.locationtech.geogig.api.plumbing.FindTreeChild;
+import org.locationtech.geogig.api.plumbing.ResolveObjectType;
+import org.locationtech.geogig.api.plumbing.ResolveTreeish;
+import org.locationtech.geogig.api.plumbing.RevObjectParse;
+import org.locationtech.geogig.api.plumbing.RevParse;
+import org.locationtech.geogig.geotools.geopkg.GeopkgAuditExport;
+import org.locationtech.geogig.geotools.plumbing.GeoToolsOpException;
+import org.locationtech.geogig.repository.Repository;
+import org.locationtech.geogig.rest.Variants;
+import org.locationtech.geogig.rest.geotools.ExportWebOp;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.restlet.data.Form;
+import org.restlet.data.Request;
+import org.restlet.resource.Variant;
+
+import com.google.common.base.Optional;
+
+/**
+ * Export Web Op for GeoPackage.
+ */
+public class GeoPkgExportWebOp extends ExportWebOp {
+
+    public static final String INTERCHANGE_PARAM = "interchange";
+
+    @Override
+    public FeatureStoreWrapper getFeatureStoreWrapper(String srcPath, String targetTable,
+        Form options) throws IOException {
+        FeatureStoreWrapper wrapper = new FeatureStoreWrapper();
+        // get a datastore
+        final GeoPkgDataStoreFactory factory = new GeoPkgDataStoreFactory();
+        final HashMap<String, Serializable> params = new HashMap<>(3);
+        // fill in DataStore parameters
+        params.put(GeoPkgDataStoreFactory.DBTYPE.key, "geopkg");
+        params.put(GeoPkgDataStoreFactory.DATABASE.key,
+            options.getFirstValue("database", "database.geopkg"));
+        params.put(GeoPkgDataStoreFactory.USER.key,
+            options.getFirstValue("user", "user"));
+        JDBCDataStore dataStore;
+        try {
+            dataStore = factory.createDataStore(params);
+        } catch (IOException ioe) {
+            throw new RuntimeException("Unable to create GeoPkgDataStore", ioe);
+        }
+        if (null == dataStore) {
+            throw new RuntimeException("Unable to create GeoPkgDataStore");
+        }
+        final File databaseFile = new File(params.get("database").toString());
+        wrapper.setBinary(databaseFile);
+        // create the Feature schema if it doesn't exist
+        ObjectId featureTypeId = createFeatureTypeSchema(dataStore, srcPath, targetTable, options);
+        // set the featureType filter on the wrapper
+        wrapper.setFeatureTypeFilterId(featureTypeId);
+        // set the SimpleFeatureStore on the wrapper
+        SimpleFeatureSource featureSource = dataStore.getFeatureSource(targetTable);
```

Comment:
this line to the end of this method is copied from the ExportDataSet code in GeoGig. It seems to be the pattern used for some of the WEB API ops, but it's terrible to just copy code like this. Maybe we can refactor it so that the copied code is reusable.